### DISCUSSION TITLE
HDR

### DISCUSSION THREAD

**Discussion by ildevblu** [MolotovCherry/virtual-display-rs#28] [Q&A] [answered]
Is it HDR supported?
If not, do you think it is something that could be added or does windows requires some kind of licensed stuff for that?

**Comment by MolotovCherry** [comment, marked as answer]
Unfortunately, as far as I know, there's no way to support it using this set of windows api

**Reply by MolotovCherry** [reply]
As an update to this, while Microsoft said this is not possible currently[^1], idd 1.10 seems to be coming with the functionality[^2]. It looks like it's possible to do in one driver using runtime checks, however the special functionality in idd 1.10 only supports Windows 11 22H2 Sept Update and higher[^3], which means this feature would only be available in the absolute newest OS (and latest update).

I have added your request in #31 , since it will be possible.

[^1]: https://github.com/MicrosoftDocs/windows-driver-docs-ddi/issues/199#issuecomment-424825422
[^2]: https://learn.microsoft.com/en-us/windows-hardware/drivers/display/iddcx1.10-updates
[^3]: https://learn.microsoft.com/en-us/windows-hardware/drivers/display/iddcx-versions#iddcx-version-info-for-windows-11-releases

**Reply by ildevblu** [reply]
Sounds cool! I think it would be the only vd existing with HDR support. You can consider to put the current release in maintenance mode in a separate branch and drop W10 and W11 22H1 support in the main branch.

**Reply by MolotovCherry** [reply]
It should be able to be done in the same driver we already have using runtime checks. Only that the feature will not be enabled if the OS is not new enough
